src/link.c

```diff
@@ -709,28 +725,39 @@ LINK_HANDLE link_create_from_endpoint(SESSION_HANDLE session, LINK_ENDPOINT_HAND
             result->role = role_sender;
         }
 
-        result->pending_deliveries = singlylinkedlist_create();
-        if (result->pending_deliveries == NULL)
+        result->tick_counter = tickcounter_create();
+        if (result->tick_counter == NULL)
         {
             free(result);
```

Comment:
>free(result); [](start = 12, length = 13)

LogError please.  Here and everywhere.

Reply:
Will be done by a separate PR, that's how I usually have them.